WordPress/Classes/Services/PostRepository.swift

```diff
@@ -63,6 +58,58 @@ final class PostRepository {
         }
     }
 
+    /// Uploads the changes to the given post to the server and deletes the
+    /// uploaded local revision afterward. If the post has an associated
+    /// remote ID, creates a new post.
+    ///
+    /// - note: This method is a low-level primitive for syncing the latest
+    /// post date to the server.
+    ///
+    /// - warning: Before publishing, ensure that the media for the post got
+    /// uploaded. Managing media is not the responsibility of `PostRepository.`
+    ///
+    /// - warning: Work-in-progress (kahu-offline-mode)
+    @MainActor
+    func _upload(_ post: AbstractPost) async throws {
+        let remotePost = PostHelper.remotePost(with: post)
+        try await _upload(remotePost, for: post)
+    }
+
+    /// Uploads the changes to the given post to the server and deletes the
+    /// uploaded local revision afterward. If the post has an associated
+    /// remote ID, creates a new post.
+    ///
+    /// - note: This method is a low-level primitive for syncing the latest
+    /// post date to the server.
+    ///
+    /// - warning: Before publishing, ensure that the media for the post got
+    /// uploaded. Managing media is not the responsibility of `PostRepository.`
+    ///
+    /// - warning: Work-in-progress (kahu-offline-mode)
+    @MainActor
+    func _upload(_ parameters: RemotePost, for post: AbstractPost) async throws {
```

Comment:
`parameters: RemotePost` – this will definitely change, but the idea is that by bringing back synchronous operations, we need a way to:

- Send the request with new parameters without updating the database to make sure there is nothing to revert if it fails
- Be able to send only what's changes and that's coming in the next scopes